src/App.tsx

```diff
@@ -1,7 +1,43 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+interface AppState {
+  pressedKey: string | null;
+}
+
+export class App extends React.Component<{}, AppState> {
+  state: AppState = {
+    pressedKey: null,
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', this.keyupHandler);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', this.keyupHandler);
+  }
+
+  keyupHandler = (event: KeyboardEvent) => {
+    this.setState({
+      pressedKey: event.key,
+    });
+  };
+
+  render() {
+    return (
+      <div className="App">
+        {this.state.pressedKey
+          ? (
+            <p className="App__message">
+              The last pressed key is [
+              {this.state.pressedKey}
+              ]
+            </p>
+          )
+          : (
+            <p className="App__message">Nothing was pressed yet</p>
+          )}
```

Comment:
You can change only the text content of `p` tag to escape duplicating it. And you can use destructuring of ` this.state` here .